### PR TITLE
AB#89099 Summary of delivery - Editable - Tranches pages

### PR DESCRIPTION
**Pull request by Carq**
image

image

### CODE REVIEW

File: src/HE.Investment.AHP.WWW/Controllers/DeliveryPhaseController.cs [hunk 1]
```diff
@@ -122,14 +125,16 @@ public async Task<IActionResult> Details([FromRoute] string applicationId, strin
...
+                new GetDeliveryPhaseDetailsQuery(AhpApplicationId.From(applicationId), new DeliveryPhaseId(deliveryPhaseId)),
+                cancellationToken);
 
         return View("Details", deliveryPhaseDetails);
     }
 
     [HttpPost("{deliveryPhaseId}/details")]
     [WorkflowState(DeliveryPhaseWorkflowState.TypeOfHomes)]
-    public async Task<IActionResult> Details([FromRoute] string applicationId, string deliveryPhaseId, DeliveryPhaseDetails deliveryPhaseDetails, CancellationToken cancellationToken)
+    public async Task<IActionResult> Details([FromRoute] string applicationId, string deliveryPhaseId, DeliveryPhaseDetails deliveryPhaseDetails,
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.WWW/Controllers/DeliveryPhaseController.cs:136]: looks like stylecop warning 🤔

File: src/HE.Investment.AHP.Contract/Delivery/MilestonePayments/Commands/ProvideAcquisitionTrancheCommand.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+using HE.Investment.AHP.Contract.Application;
+using HE.Investment.AHP.Contract.Delivery.Commands;
+using HE.Investments.Common.Contract.Validators;
+using MediatR;
+
+namespace HE.Investment.AHP.Contract.Delivery.MilestonePayments.Commands;
+
+public record ProvideAcquisitionTrancheCommand(AhpApplicationId ApplicationId, DeliveryPhaseId DeliveryPhaseId, decimal? AcquisitionTranche)
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Contract/Delivery/MilestonePayments/Commands/ProvideAcquisitionTrancheCommand.cs:8]: Where is min/max validation for Tranche? Can user enter non numeric characters?

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.Contract/Delivery/MilestonePayments/Commands/ProvideAcquisitionTrancheCommand.cs (hunk 2 above)
Comment (Carq) [reply src/HE.Investment.AHP.Contract/Delivery/MilestonePayments/Commands/ProvideAcquisitionTrancheCommand.cs:8]: It will be part of next PR.

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Views/DeliveryPhase/SummaryOfDeliveryTranche.cshtml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    ViewData["Title"] = title;
+}
+
+<gds-div-grid-column-two-thirds>
+    <vc:ahp-workflow-back-button current-page="@DeliveryPhaseWorkflowState.SummaryOfDeliveryTranche"/>
+
+    <form asp-controller="DeliveryPhase" asp-action="SummaryOfDeliveryTranche" asp-method="post" asp-route-applicationId="@applicationId" asp-route-deliveryPhaseId="@Model.DeliveryPhaseId" asp-route-redirect="@Context.Request.Query["redirect"]" novalidate>
+        <vc:form-header caption="@($"{Model.ApplicationName} - {Model.DeliveryPhaseName}")" title="@title" hint="@description"/>
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Views/DeliveryPhase/SummaryOfDeliveryTranche.cshtml:43]: Is error message for this case displayed in valid way?
Looks like is not.

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Views/DeliveryPhase/SummaryOfDeliveryTranche.cshtml (hunk 3 above)
Comment (Carq) [reply src/HE.Investment.AHP.WWW/Views/DeliveryPhase/SummaryOfDeliveryTranche.cshtml:43]: It will be fixed in the PR.

**Pull request merged**